ament_copyright/test/test_parser.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ament_copyright.parser import search_copyright_information
+
+
+def test_search_copyright_information_incorrect_typo():
+    """Test searching for copyright information with a typo in the copyright information."""
+    copyrights, remaining_block = search_copyright_information(
+        'CopyrightTypo 2020 Open Source Robotics Foundation, Inc.'
+    )
+    assert len(copyrights) == 0
+
+
+def test_search_copyright_information_repeated():
+    """Test searching with repeated 'copyright' in the copyright information."""
+    copyrights, remaining_block = search_copyright_information(
+        '\\copyright Copyright 2020 Open Source Robotics Foundation, Inc.'
+    )
+    assert len(copyrights) > 0
```

Comment:
This is probably better since it's more specific and it's what you expect, correct? This applies to other spots where you do the same thing.
```suggestion
    assert len(copyrights) == 2
```